### PR TITLE
Deprecate Kotest 4 and promote Kotest 5 to managed dependencies

### LINKED ISSUE #604: Deprecate Kotest 4 and promote Kotest 5 to managed dependencies
### Issue description

The Kotest 4 dependencies should be deprecated under `gradle/libs.versions.toml`. The Kotest 5 dependencies should be promoted to managed dependencies. This is needed to fully support the following:

- https://github.com/micronaut-projects/micronaut-starter/issues/1216
- https://github.com/micronaut-projects/micronaut-gradle-plugin/pull/483

### PR DESCRIPTION
**Pull request by wetted**
closes #604

### CODE REVIEW

**Review by timyates — APPROVED** [review]
I think this is the best we can do 👍

If only they'd changed the group or artifact id

I guess this needs to wait for Micronaut 4, as it will break anyone using Kotest and the catalog

(admittedly probably a small subset of users)

Unless we also add `kotest-assertions-core-jvm` and `kotest-runner-junit5-jvm` to the core catalog 🤔

**Comment by timyates** [comment]
Wondering if we should leave these two as managed

```
managed-kotest-assertions-core-jvm
managed-kotest-runner-junit5-jvm
```

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -22,10 +22,10 @@ managed-hamcrest = { module = "org.hamcrest:hamcrest", version.ref = "managed-ha
 managed-mockito-core = { module = "org.mockito:mockito-core", version.ref = "managed-mockito" }
 managed-mockito-junit-jupiter = { module = "org.mockito:mockito-junit-jupiter", version.ref = "managed-mockito" }
 managed-mockk = { module = "io.mockk:mockk", version.ref = "managed-mockk" }
-managed-kotest-assertions-core-jvm = { module = "io.kotest:kotest-assertions-core-jvm", version.ref = "managed-kotest" }
-managed-kotest-runner-junit5-jvm = { module = "io.kotest:kotest-runner-junit5-jvm", version.ref = "managed-kotest" }
```
Comment (timyates) [inline gradle/libs.versions.toml:26]: Can we leave these as managed?

**Review by timyates — CHANGES_REQUESTED** [review]

File: gradle/libs.versions.toml (hunk 1 above)
Comment (timyates) [reply gradle/libs.versions.toml:26]: Ignore me, this would break the POM

**Comment by timyates** [comment]
@wetted @sdelamo Pushed a change so we re-use existing catalog names, but now kotest5 is managed

I think it will simplify things in core, but this needs to be a major release here, and Micronaut 4.0.0

**Review by timyates — APPROVED** [review]

**Comment by wetted** [comment]
@timyates thanks, I agree this will have to be a 4.0.x feature, for potential breaking changes to managed dependencies.

**Pull request merged**